src/Features/LanguageServer/CommonLanguageServerProtocol.Framework/IQueueItem.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace CommonLanguageServerProtocol.Framework;
+
+#nullable enable
+
+public interface IQueueItem<RequestContextType>
+{
+    /// <summary>
+    /// Begins executing the work specified by this queue item.
+    /// </summary>
+    Task CallbackAsync(RequestContextType? context, CancellationToken cancellationToken);
```

Comment:
`CallbackAsync` feels odd, maybe `StartRequestAsync`?

Reply:
agreed.  Also, can we doc when 'context' is null?